### PR TITLE
Reviewed error context usage

### PR DESCRIPTION
**Pull request by naz**
refs #10571

Reduced and improved error handling in places where `context` was polluting the output. @ErisDS there are couple questions around these changes will leave them in the code comments :+1:

### CODE REVIEW

File: core/server/models/plugins/collision.js [hunk 1]
```diff
@@ -63,10 +63,9 @@ module.exports = function (Bookshelf) {
                                     message: 'Saving failed! Someone else is editing this post.',
                                     code: 'UPDATE_COLLISION',
                                     level: 'critical',
-                                    context: {
+                                    errorDetails: {
                                         clientUpdatedAt: self.clientData.updated_at,
-                                        serverUpdatedAt: self.serverData.updated_at,
-                                        changed: changed
```
Comment (gargol) [inline core/server/models/plugins/collision.js]: The `change` field was introduced temporarily here - https://github.com/TryGhost/Ghost/commit/30e790bf122da291d2b27f52cdde767a76386132#diff-eb942d2a7406b236df44c35a83722627 but then the comment was removed in https://github.com/TryGhost/Ghost/commit/3289dc76191ce90c3d1149610482134110171164#diff-eb942d2a7406b236df44c35a83722627 . Think we don't need to keep it here anymore.

File: core/server/services/themes/validate.js [hunk 2]
```diff
@@ -31,8 +33,11 @@ checkTheme = function checkTheme(theme, isZip) {
 
             return Promise.reject(new common.errors.ThemeValidationError({
                 message: common.i18n.t('errors.api.themes.invalidTheme'),
-                errorDetails: checkedTheme.results.error,
-                context: checkedTheme
+                errorDetails: Object.assign(
+                    _.pick(checkedTheme, ['checkedVersion', 'name', 'path', 'version']), {
```
Comment (gargol) [inline core/server/services/themes/validate.js]: Only left some useful fields as the rest is just noise

File: core/server/services/themes/validate.js [hunk 3]
```diff
@@ -31,7 +33,11 @@ checkTheme = function checkTheme(theme, isZip) {
 
             return Promise.reject(new common.errors.ThemeValidationError({
                 message: common.i18n.t('errors.api.themes.invalidTheme'),
-                errorDetails: checkedTheme.results.error,
+                errorDetails: Object.assign(
+                    _.pick(checkedTheme, ['checkedVersion', 'name', 'path', 'version']), {
+                        errors: checkedTheme.results.error
+                    }
+                ),
                 context: checkedTheme
```
Comment (gargol) [inline core/server/services/themes/validate.js]: Unfortunately, this whole object has to stay here for now, and [the flow](https://github.com/TryGhost/Ghost/blob/9810834/core/server/services/themes/index.js#L56-L57) that depends on `context` being set to `checkedTheme` needs more investigation :disappointed:

**Review by ErisDS — COMMENTED** [review]
I don't see any questions?

File: core/server/models/plugins/collision.js (hunk 1 above)
Comment (kirrg001) [reply core/server/models/plugins/collision.js]: We've added the changed information for transparency and debugging purposes.

**Comment by gargol** [comment]
@ErisDS sorry, my questions became just "comments" in the end :sweat_smile: 
Summing up this PR: the removal of `changed` field is ok, `checkedTheme` will need more investigation next Monday.

**Pull request merged**